src/fastmcp/server/server.py

```
@@ -998,13 +1030,14 @@ async def get_component(
-            except NotFoundError:
-                continue
+        results = await gather(
+            *[p.get_component(key) for p in self._providers],
+            return_exceptions=True,
+        )
+
+        for result in results:
+            if isinstance(result, FastMCPComponent):
+                return result
 
         raise NotFoundError(f"Unknown component: {key}")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)